### PR TITLE
[#49076] Add technical documentation for file storage integration

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/49076
- added check list for new file storage integration candidates

### CODE REVIEW

**Comment by Kharonus** [comment]
@wielinde please definitely check, if the location of the checklist is okay, or if we should merge it into any existing docs.

File: docs/development/file-storage-integration/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# File storage integration
```
Comment (wielinde) [inline docs/development/file-storage-integration/README.md]:
```suggestion
# Integrating more external file storage providers
```

File: docs/development/file-storage-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+External file storages can be used from within OpenProject with the file storage module. This adds certain use cases,
```
Comment (wielinde) [inline docs/development/file-storage-integration/README.md]:
```suggestion
External file storages can be integrated with OpenProject via the _File storages_ module. This adds certain use cases,
```

File: docs/development/file-storage-integration/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+e.g. in the files tab of a work package, to interact with files located on the storage and link them to entities in
```
Comment (wielinde) [inline docs/development/file-storage-integration/README.md]:
```suggestion
e.g. in the _Files_ tab of a work package, to interact with files located on the storage and link them to work packages in
```

**Review by wielinde — CHANGES_REQUESTED** [review]
@Kharonus Looks good to me so far. Maybe we could add some differentiation to attachments and their direct uploads, explaining that this is not about that? Also please check my inline comments/suggestions.

I don't really understand what will be the title of the menu item in the docs for this page.

**Comment by Kharonus** [comment]
The title for the menu is something, I don't know yet. This is also a reason, why I asked, if we should integrate it somewhere else. Will first add your changes.

File: docs/development/file-storage-integration/README.md (hunk 1 above)
Comment (Kharonus) [reply docs/development/file-storage-integration/README.md]: added

File: docs/development/file-storage-integration/README.md (hunk 2 above)
Comment (Kharonus) [reply docs/development/file-storage-integration/README.md]: added

File: docs/development/file-storage-integration/README.md (hunk 3 above)
Comment (Kharonus) [reply docs/development/file-storage-integration/README.md]: added

**Comment by Kharonus** [comment]
@wielinde Can we finish that list up? Open question is currently, where to put that list.

**Review by wielinde — APPROVED** [review]

**Comment by wielinde** [comment]
@Kharonus I added a couple of changes. If you are fine with them, please go ahead and merge.

**Pull request merged**